hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -332,6 +335,8 @@ public AbfsRestOperation setFilesystemProperties(final String properties,
     final AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
     abfsUriQueryBuilder.addQuery(QUERY_PARAM_RESOURCE, FILESYSTEM);
 
+    appendSASTokenToQuery(ROOT_PATH, "", abfsUriQueryBuilder);
```

Comment:
why operationName is not provided here. If the thing is that for container APIs, existing SAS mechanism can not work, we should still prevent them. We might have to add intelligence which SAS implementations are allowed for container APIs.

Reply:
Nice Catch. Operations name added for file system level operations as well.

Just FYI
The purpose of having operation name here is to determine what all permissions are needed to be set in SAS Token generated. It is up to the custom SAS Token Provider Implementation to use this information and set appropriate permissions. For Example, MockDelegationSASTokenProvider does not allow file system level operations hence it will error out for these operations with SASTokenProviderException.

Where as AccountSASGenerator will have fixed permissions irrespective of operation type.

Operation Name are added here so that if a user wants to define their own implementation, they can choose to consume this information as per their needs and logic. They can also choose to simply ignore them.

Reply:
`appendSASTokenToQuery` was not there before this patch. Is this added for fixed sas token only. If yes, then we would have to add if-condition to keep only fixedSasToken (if in config) applying to the container APIs, and in non-fixedSasToken case, `getSASToken` should not be called. Reason being, developers would already have their implementations and those might not be adaptable to the container APIs.

Reply:
As discussed, we don't want to support any container API on SAS Token Authentication. Removed these changes